test/server/admin/stats_html_render_test.cc

```diff
@@ -38,12 +38,22 @@ TEST_F(StatsHtmlRenderTest, String) {
               HasSubstr("name: \"abc 123 ~!@#$%^&amp;*()-_=+;:&#39;&quot;,&lt;.&gt;/?\"\n"));
 }
 
-TEST_F(StatsHtmlRenderTest, HistogramNoBuckets) {
+TEST_F(StatsHtmlRenderTest, HistogramUnset) {
```

Comment:
Can you factor out the body of this test to a helper function so we can share the code between HistogramUnset and HistogramSummary? I think we should also test the compatibility of "none" in a third call to that helper.

Reply:
I'm not sure how much this would actually buy us, as the tests are pretty slim as is. Also, I don't think testing "none" is important here, since the HTML renderer should only care about the possible bucket modes, not the query params.

Reply:
I think factoring out the test body makes it more obvious that the Unset and Summary have the same behavior, without having to read and compare a few lines of numbers.

But you are right about not being able to test 'none' because this test specifies an enum, not a query-param, and you have the parsing test covered in utils_test.cc.